### PR TITLE
Separate out the 'band selection' code

### PR DESCRIPTION
**Pull request by kholia**
Changes:

- Separate out the 'band selection' code
- Format files using Arduino's auto-format option
- Improve Makefile to format code on-demand
- Call 'band selection' code from FT-817 code (still in progress)
- The `formatter.conf` file comes from upstream Arduino project itself
- Unbreak CAT once more!

### CODE REVIEW

File: ADX FIRMWARE/EXPERIMENTAL/PDX_V2.0/ft817.cpp [hunk 1]
```diff
@@ -229,6 +232,42 @@ void processCATCommand2(byte* cmd) {
...
+      // If a band change is detected switch to the new band
+      if (b != Band_slot) { // band change
+        Band_slot = b;
+        Freq_assign();
+      }
+      // Properly register the mode if the frequency implies a WSJT mode change
+      // (FT8,FT4,JS8,WSPR)
+      i = getBand(freq);
+      if (i < 0) {
+        break;
```
Comment (kholia) [inline ADX FIRMWARE/EXPERIMENTAL/PDX_V2.0/ft817.cpp:245]: A `break` here doesn't make sense (code flow-wise) but it was there in the original code for some reason.

Comment (lu7did) [reply ADX FIRMWARE/EXPERIMENTAL/PDX_V2.0/ft817.cpp:244]: Perhaps this is what is breaking the communications. Will come later if needed. For now I am flowing in the merge

**Comment by lu7did** [comment]
Done

File: ADX FIRMWARE/EXPERIMENTAL/PDX_V2.0/ft817.cpp (hunk 1 above)
Comment (kholia) [reply ADX FIRMWARE/EXPERIMENTAL/PDX_V2.0/ft817.cpp:244]: Ack - I will look out for some code sections.

**Pull request merged**